### ISSUE TITLE
DoesNotReturn analysis doesn't complain about `return;`

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#45791]
**Version Used**: 7b4bfaaee

**Steps to Reproduce**:

```cs
using System.Diagnostics.CodeAnalysis;

#nullable enable

public class C {
    [DoesNotReturn]
    public void M() {
        return;
    }
}
```

**Expected Behavior**: Warning on `return;`

**Actual Behavior**: No warnings. Deleting the `return;` statement results in a `warning CS8763: A method marked [DoesNotReturn] should not return.` on the closing brace.

Root cause is likely that we bail out of return statement analysis before checking DoesNotReturn when there is no return value.
https://github.com/dotnet/roslyn/blob/be56bf44b7bcd2462c123aca88282791b4e5bca0/src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs#L1961
